### PR TITLE
Fix GPU-autofill for rayStartParams

### PR DESCRIPTION
**Pull request by DmitriGekhtman**
## Why are these changes needed?

Closes https://github.com/ray-project/kuberay/issues/321

Fixes the logic that appends --num-gpus tp rayStartParams based on container resources.

The issue is that current logic looks for a resource named "gpu". However device plugins require extended resource names like "nvidia.com/gpu". 

This PR tweaks the logic so that we search for a resource key ending with ~~"/gpu"~~ "gpu".

## Checks

- [ ] I've made sure the tests are passing. 
- Testing Strategy
   - [x] Unit tests
   - [ ] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: ray-operator/controllers/ray/common/pod.go [hunk 1]
```diff
@@ -411,9 +411,13 @@ func concatenateContainerCommand(nodeType rayiov1alpha1.RayNodeType, rayStartPar
 	}
 
 	if _, ok := rayStartParams["num-gpus"]; !ok {
-		gpu := resource.Limits["gpu"]
-		if !gpu.IsZero() {
-			rayStartParams["num-gpus"] = strconv.FormatInt(gpu.Value(), 10)
+		// Scan for resource keys ending with "/gpu" like "nvidia.com/gpu".
+		for resourceKey, resource := range resource.Limits {
+			if strings.HasSuffix(string(resourceKey), "/gpu") && !resource.IsZero() {
```
Comment (akanso) [inline ray-operator/controllers/ray/common/pod.go:416]: ideally, we would want "/gpu", "memory", "num-cpu" etc. in a separate file as constants, and that way if they change, we wouldn't have to change them in every place where they are used.

This could be a different PR that does that.

**Review by akanso — APPROVED** [review]

File: ray-operator/controllers/ray/common/pod.go (hunk 1 above)
Comment (DmitriGekhtman) [reply ray-operator/controllers/ray/common/pod.go:416]: Good idea -- will add some constants in a follow-up.

Comment (sriram-anyscale) [reply ray-operator/controllers/ray/common/pod.go:416]: Is it always going to end with "/gpu"? Or is it ever possible for it to be just "gpu"?

**Review by sriram-anyscale — APPROVED** [review]

File: ray-operator/controllers/ray/common/pod.go (hunk 1 above)
Comment (pcmoritz) [reply ray-operator/controllers/ray/common/pod.go:416]: +1 to Sriram's comment -- what was the previous behavior here (checking resource.Limits["gpu"]), why was it implemented but not unit-tested and are we making breaking changes here?

Presumably if it was not tested we can change it, but why was it there in the first place?

**Review by pcmoritz — APPROVED** [review]

File: ray-operator/controllers/ray/common/pod.go (hunk 1 above)
Comment (DmitriGekhtman) [reply ray-operator/controllers/ray/common/pod.go:416]: GPUs are always specified as extended resources. In practice, either "nvidia.com/gpu" or "amd.com/gpu".
It appears to have been an oversight. Technically, dropping the "/" from the suffix will keep backwards compatibility, so I'll do that.
